### PR TITLE
User should be able to browse public tutorials and click on links #27

### PR DESCRIPTION
**Pull request by bbiehl**

### CODE REVIEW

File: app/controllers/searches_controller.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class SearchesController < ApplicationController
   def index
-    @results = Tutorial.where(:public => true, :draft => false).search_by_title(params[:q])
+    @results = Tutorial.where(:public => true, :draft => false).search_by_title(params[:q]) || Tutorial.where(:public => true, :draft => false).search_by_description
```
Comment (sumeetjain) [inline app/controllers/searches_controller.rb]: Calls tutorial.where(:public => true, :draft => false) 2 times (DuplicateMethodCall)

Comment (sumeetjain) [inline app/controllers/searches_controller.rb]: Use the new Ruby 1.9 hash syntax.

Comment (sumeetjain) [inline app/controllers/searches_controller.rb]: Line is too long. [165/80]

**Comment by bbiehl** [comment]
After testing, this is not searching the tutorial description.  In which Sumeet alluded to earlier.

File: app/controllers/searches_controller.rb [hunk 2]
```diff
@@ -1,5 +1,19 @@
 class SearchesController < ApplicationController
+  def title_matches
+    @tutorials = Tutorial.where(public: true, draft: false)
+    @title_matches = @tutorials.search_by_title(params[:q])
+  end
+
+  def description_matches
+    @tutorials = Tutorial.where(public: true, draft: false)
+    @description_matches = @tutorials.search_by_description(params[:q])
```
Comment (sumeetjain) [inline app/controllers/searches_controller.rb]: Remove trailing whitespace

Comment (sumeetjain) [inline app/controllers/searches_controller.rb]: Trailing whitespace detected.

File: app/controllers/searches_controller.rb [hunk 3]
```diff
@@ -1,5 +1,19 @@
...
+  end
+
+  def description_matches
+    @tutorials = Tutorial.where(public: true, draft: false)
+    @description_matches = @tutorials.search_by_description(params[:q])    
+  end
+
   def index
-    @results = Tutorial.where(:public => true, :draft => false).search_by_title(params[:q])
+    if title_matches.length > 0 || description_matches.length > 0
```
Comment (sumeetjain) [inline app/controllers/searches_controller.rb]: Use `!empty?` instead of `length > 0`.

File: app/controllers/searches_controller.rb [hunk 4]
```diff
@@ -1,5 +1,19 @@
...
+      @tutorials = Tutorial.where(public: true, draft: false)
+      @results = @tutorials.search_by_title(params[:q])
+    else
```
Comment (sumeetjain) [inline app/controllers/searches_controller.rb]: Redundant `else`-clause.

File: app/controllers/searches_controller.rb [hunk 5]
```diff
@@ -1,5 +1,18 @@
...
+
+  def description_matches
+    @description_matches = @tutorials.search_by_description(params[:q])
+  end
+
   def index
-    @results = Tutorial.where(:public => true, :draft => false).search_by_title(params[:q])
+    @tutorials = Tutorial.where(public: true, draft: false)
+    if title_matches.empty? == false
+      @results = @tutorials.search_by_title(params[:q])
```
Comment (sumeetjain) [inline app/controllers/searches_controller.rb]: Calls params[:q] 2 times (DuplicateMethodCall)

File: app/controllers/searches_controller.rb [hunk 6]
```diff
@@ -1,5 +1,18 @@
...
+    else description_matches.empty? == false
```
Comment (sumeetjain) [inline app/controllers/searches_controller.rb]: Odd `else` layout detected. Did you mean to use `elsif`?

Comment (sumeetjain) [inline app/controllers/searches_controller.rb]: Operator `==` used in void context.

File: app/controllers/searches_controller.rb [hunk 7]
```diff
@@ -1,5 +1,18 @@
...
+      @results = @tutorials.search_by_description(params[:q])
```
Comment (sumeetjain) [inline app/controllers/searches_controller.rb]: Inconsistent indentation detected.

**Comment by bbiehl** [comment]
We're not actually searching any descriptions. 
<code>We get undefined method `search_by_description' for #Tutorial::ActiveRecord_Relation:0x007fd0f633e888 Did you mean? search_by_title</code>

**Comment by bbiehl** [comment]
- [x] We want to display only the title of a public tutorial.  This we can do.
- [x] If a search query matches a keyword in a tutorial's title, we want that tutorial link to appear as its title in a search result.
- [ ] If a search query matches a keyword in a tutorial's title, we also don't need to search that tutorial's description.
- [ ] If a search query doesn't match a keyword in a tutorial's title, we then want to search that tutorial's description.
- [ ] If a search query doesn't match a keyword in a tutorial's title,  but does match a keyword in that tutorial's description, we want that tutorial's link to appear as its title in a search result.

**Comment by bbiehl** [comment]
Correction.  Only returns the tutorial.title of the second row of my database.  Regardless of what the search term is.

File: app/controllers/searches_controller.rb [hunk 8]
```diff
@@ -1,5 +1,7 @@
 class SearchesController < ApplicationController
   def index
-    @results = Tutorial.where(:public => true, :draft => false).search_by_title(params[:q])
+    @tutorials = Tutorial.where(public: true, draft: false)
+    # @results = @tutorials.search_by_title(params[:q])
+    @results = @tutorials.where("title LIKE ? or description LIKE ?", "%#{:q}%", "%#{:q}%")
```
Comment (sumeetjain) [inline app/controllers/searches_controller.rb]: Literal interpolation detected.

Comment (sumeetjain) [inline app/controllers/searches_controller.rb]: Line is too long. [91/80]

File: app/models/tutorial.rb [hunk 9]
```diff
@@ -2,6 +2,7 @@ class Tutorial < ActiveRecord::Base
   has_many :chapters, dependent: :destroy
   include PgSearch
   pg_search_scope :search_by_title, :against => :title
+  pg_search_scope :search_by_description, :against => :description
```
Comment (sumeetjain) [inline app/models/tutorial.rb]: Use the new Ruby 1.9 hash syntax.

**Pull request merged**